hoprd/rest-api/src/messages.rs

```diff
@@ -143,42 +143,47 @@ pub(super) async fn send_message(
         .map(|enc| enc(&args.body))
         .unwrap_or_else(|| args.body.into_boxed_slice());
 
-    let path = args.path;
-
     // TODO (jean): find a nice way to handle PeeroOrAddress fulfilling
-    // if let Some(path) = path {
-    //     let path = path
-    //         .iter()
-    //         .map(|address| {
-    //             let hopr_db = hopr.hopr_db();
-    //             async move { address.clone().fulfill(&hopr_db).await }
-    //         })
-    //         .collect::<Vec<_>>();
-
-    //     let path = futures::future::join_all(path).await;
-    //     if path.iter().any(|p| p.is_err()) {
-    //         return Err((StatusCode::NOT_FOUND, ApiErrorStatus::InvalidInput).into_response());
-    //     }
-    // }
-
-    let options = if let Some(intermediate_path) = path {
+
+    let options = if let Some(intermediate_path) = args.path {
+        let path_futures = intermediate_path
+            .into_iter()
+            .map(|address| address.fulfill(hopr.peer_resolver()))
```

Comment:
You could do smth like :

`address.fulfill(hopr.peer_resolver()).and_then(|resolved| resolver.peer_id)`

here and then discard the entire block on lines 162-177.